### PR TITLE
Discovery refactor

### PR DESCRIPTION
**Pull request by prashanthlgowda**
Discovery refactor

### CODE REVIEW

**Comment by michaelsteven** [comment]
Resolve conflicts first.

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceDefaultCredential.java [hunk 1]
```diff
@@ -1,122 +1,73 @@
-/**
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceDefaultCredential.java]: Did you really intend to remove this header and the javadoc?

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceGroup.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceGroup.java:4]: Missing file header

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryConfig.java [hunk 3]
```diff
@@ -12,78 +11,51 @@
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
-/**
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryConfig.java]: Did you really mean to remove the javadoc?

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryRule.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
+
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryRule.java]: Missing copyright header

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/Identifier.java [hunk 5]
```diff
@@ -11,78 +10,51 @@
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
-/**
```
Comment (michaelsteven) [inline src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/Identifier.java]: Did you really mean to remove the javadoc?

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceDefaultCredential.java (hunk 1 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceDefaultCredential.java]: These are new generated objects.

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceGroup.java (hunk 2 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceGroup.java:4]: These are new generated objects.

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryConfig.java (hunk 3 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryConfig.java]: These are new generated objects.

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryRule.java (hunk 4 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryRule.java]: These are new generated objects.

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/Identifier.java (hunk 5 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/Identifier.java]: These are new generated objects.

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceGroup.java (hunk 2 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DeviceGroup.java:4]: modified now

File: src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryRule.java (hunk 4 above)
Comment (prashanthlgowda) [reply src/main/java/com/dell/isg/smi/commons/model/device/discovery/config/DiscoveryRule.java]: modified now

**Pull request merged**